### DISCUSSION TITLE
Private key password supports encryption

### DISCUSSION THREAD

**Discussion by 731197363** [minio/minio#17874] [Q&A] [answered]
rerfence： https://github.com/minio/minio/blob/master/docs/tls/README.md#install-certificates-from-third-party-cas
![image](https://github.com/minio/minio/assets/10794822/43129a77-e149-472e-b40b-b7fba0a5c84e)

> export MINIO_CERT_PASSWD=PASSWORD

when enable ssl.  start minio running,  the private key  password will show clear-text password in the enviorment. 

will minio support encrypted password In future versions

**Comment by klauspost** [comment]
How would you decrypt an encrypted password?

**Reply by 731197363** [reply]
yes，It's a foolish and  idea

shall we follow the way like nginx ：

 when I reload nginx by doing service nginx reload, I keep getting this prompted:

`Reloading nginx configuration: Enter PEM pass phrase:`


just make it  interactive way to input the password .

**Reply by klauspost** [reply, marked as answer]
This makes sense for interactive applications but not for servers.

We don't really add features that give users a big opportunity to shoot themselves in the foot.

**Reply by 731197363** [reply]
thanks for your patience

**Reply by klauspost** [reply]
No problem 👍🏼
